### PR TITLE
Installation guide for on-premise cluster

### PR DESCRIPTION
**Pull request by elviraux**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1739)

### CODE REVIEW

File: content/docs/onpremise/deploy.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+You can do that by checking that following flags are set:
+```bash
+  --service-account-issuer
+  --service-account-signing-key-file
+  --service-account-api-audiences
+```
+
+If the verification has failed you will need to set up APIServer flags for Istio Secure Gateways (SDS). Follow istio KinD Kubefadm configs [here](https://github.com/istio/istio/blob/master/prow/config/trustworthy-jwt-13-14.yaml). To read more about Istio Secure Gateways follow the [link](https://istio.io/blog/2019/trustworthy-jwt-sds/).
+
+* *Set up `kfctl` command on your cluster*:
```
Comment (swiftdiaries) [inline content/docs/onpremise/deploy.md]: The codetags on kfctl makes it look too light against the white background

File: content/docs/onpremise/deploy.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+weight = 1
++++
+
+This guide is here to help you to deploy Kubeflow on vanilla kubernetes cluster
+
+## Before you start
+
+* *Verify that Istio is supported in the current Kubernetes environment*.
+
+You can do that by checking that following flags are set:
```
Comment (swiftdiaries) [inline content/docs/onpremise/deploy.md]: Small nit: following <API Server> flags are set:

File: content/docs/onpremise/deploy.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+cd ${KFAPP}
+wget -O kfctl_k8s.yaml https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_k8s_istio.yaml
+kfctl build -V -f kfctl_k8s.yaml
+kfctl apply -V -f kfctl_k8s.yaml
+```
+
+*Multi-tenant installation:*
+
+```bash
+export KFAPP=kf-k8s
```
Comment (swiftdiaries) [inline content/docs/onpremise/deploy.md]: kf-k8s -> kf-dex

**Comment by elviraux** [comment]
@jlewi need your review

**Comment by elviraux** [comment]
Preview link https://deploy-preview-1739--competent-brattain-de2d6d.netlify.com/docs/onpremise/deploy/

File: content/docs/onpremise/_index.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
++++
+title = "Kubeflow on-Premises"
```
Comment (sarahmaddox) [inline content/docs/onpremise/_index.md]: Let's standardize the capitalization as "On-premises" when used in a page title.

File: content/docs/onpremise/deploy.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
++++
+title = "Deployment On-premises"
+description = "Instructions to deploy Kubeflow on-premise cluster"
```
Comment (sarahmaddox) [inline content/docs/onpremise/deploy.md]: Let's standardize to `on-premises` (with the final "s") wherever we use this term. 

There's a lot of debate about the term currently. In the US, many people say "on-premise" but outside the US that sounds odd to most people. A *premise* is an assumption, whereas *premises* refers to a building, location, or site. The Google developer's style guide, which we follow in the Kubeflow docs, recommends "on-premises".

File: content/docs/onpremise/deploy.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
++++
+title = "Deployment On-premises"
```
Comment (sarahmaddox) [inline content/docs/onpremise/deploy.md]: Change the directory name from `onpremise` to `on-premises`. There are two reasons for this suggestion. Firstly, see my comment below about the terms *premise* and *premises*. Secondly, it's better for SEO to separate words with hyphens in URLs.

**Comment by elviraux** [comment]
@sarahmaddox thank you for the review.
as I renamed the folder the preview link has been changed to https://deploy-preview-1739--competent-brattain-de2d6d.netlify.com/

**Comment by sarahmaddox** [comment]
@elviraux Now that I've had the opportunity to look at this in more detail, I'm wondering how this doc should interact with the existing docs here:
https://www.kubeflow.org/docs/started/k8s/

Should we perhaps change the title of the existing doc section, from **Kubernetes Installation** to something like **Installing Kubeflow on Kubernetes On-premises or in the Cloud**?

And then, should the info in the new [on-premises page](https://deploy-preview-1739--competent-brattain-de2d6d.netlify.com/docs/on-premises/deploy/) be merged into this existing doc instead?
https://www.kubeflow.org/docs/started/k8s/overview/?

**Comment by elviraux** [comment]
@sarahmaddox you are right! We definitely need to work on our website structure at Kubeflow summit. Let me then review those two pages and see if they are missing some updates. I close the pr for now. It can be reopened later on if needed

**Pull request closed without merging**